python/az/aro/azext_aro/_aad.py

```diff
@@ -54,6 +56,17 @@ def get_service_principal(self, app_id):
         return None
 
     def create_service_principal(self, app_id):
-        return self.client.service_principals.create(ServicePrincipalCreateParameters(
-            app_id=app_id,
-        ))
+        service_principal = None
+        retries = 0
+        while True:
+            try:
+                service_principal = self.client.service_principals.create(
+                    ServicePrincipalCreateParameters(app_id=app_id))
+                break
+            except GraphErrorException:
+                if retries >= 3:
+                    raise
+                retries += 1
+                time.sleep(3)
```

Comment:
Not sure about the duration of the sleep or the number of retries

Reply:
3 retries at 10 seconds each?

Reply:
That's what it was originally :smile: https://github.com/Azure/ARO-RP/pull/631#discussion_r417011667

I can change it back